onnxruntime/core/framework/graph_partitioner.cc

```diff
@@ -216,7 +206,14 @@ static Node* PlaceNode(Graph& graph, const IndexedSubGraph& capability,
         std::string node_name = oss.str();
 
         Node* fused_node = nullptr;
-        if (fusion_style == IExecutionProvider::FusionStyle::Function) {
+        // TODO: The DML currently use some legacy approach.
+        // It registers a generic predefined kernel for all purpose fusion,
+        // so it rely on the function body in the fused node during kernel creation,
+        // which is after the graph partition phase.
+        // Ideally, it should be moved to "Compile" call.
+        // Here we temporary keep the function body for DML fusion
+        // Need to remove it after migrate DML to the Compile-based approach.
+        if (provider_type == kDmlExecutionProvider) {
           fused_node = &graph.FuseSubGraph(capability, node_name);
         } else {
```

Comment:
If we hit the special case for DML do we need the old behavior of looking up the kernel in the KernelRegistryManager so there's no Compile call later?